### PR TITLE
FIX: Dont process xpub if wallet is multisig

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/a6kr776f5z0tx92r5y1vtnwbzg

[download apk](http://filestorage.bluewallet.io:1488/ref-1618982924.apk)

File: WatchConnectivity.ios.js [hunk 1]
```diff
@@ -116,100 +116,102 @@ function WatchConnectivity() {
...
+
+    for (const wallet of wallets) {
+      let receiveAddress;
+      if (wallet.getAddressAsync) {
+        if (wallet.chain === Chain.ONCHAIN) {
+          try {
+            receiveAddress = await wallet.getAddressAsync();
+          } catch (_) {}
+          if (!receiveAddress) {
+            // either sleep expired or getAddressAsync threw an exception
```
Comment (Overtorment) [inline WatchConnectivity.ios.js:129]: misleading comment, we dont hae Promise.race anywhere near..?

File: WatchConnectivity.ios.js [hunk 2]
```diff
@@ -116,100 +116,102 @@ function WatchConnectivity() {
...
+            receiveAddress = wallet._getExternalAddressByIndex(wallet.next_free_address_index);
+          }
+        } else if (wallet.chain === Chain.OFFCHAIN) {
+          try {
+            await wallet.getAddressAsync();
+            receiveAddress = wallet.getAddress();
+          } catch (_) {}
+          if (!receiveAddress) {
+            // either sleep expired or getAddressAsync threw an exception
```
Comment (Overtorment) [inline WatchConnectivity.ios.js:138]: same

**Review by Overtorment — APPROVED** [review]

File: WatchConnectivity.ios.js (hunk 1 above)
Comment (marcosrdz) [reply WatchConnectivity.ios.js:129]: It was a copy paste from Receive screen. Ill remove it

Comment (Overtorment) [reply WatchConnectivity.ios.js:129]: and maybe you do need to port `Promise.race` as well so there is a timeout for the operation..? not sure

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**